source/TimePicker.js

```diff
@@ -53,14 +53,73 @@
 		published: {
 			/**
 			* The meridiem text to display if [meridiemEnable]{@link moon.TimePicker#meridiemEnable}
-			* is `true`. The first item is used if the `hour` is less than `11`; otherwise, the
+			* is `true`. The first item is used if the `hour` is less than `12`; otherwise, the
 			* second is used.
 			*
 			* @type {String[]}
-			* @default ['AM','PM']
+			* @default [{name: 'AM', start: '00:00', end: '11:59'}, {name: 'PM', start: '12:00', end: '23:59'}]
 			* @public
 			*/
-			meridiems: ['AM','PM']
+			meridiems: [{name: 'AM', start: '00:00', end: '11:59'}, {name: 'PM', start: '12:00', end: '23:59'}]
+		},
+
+		/**
+		* @private
+		*/
+		initILib: function() {
+			// Get localized meridiem values
+			var fmtParams = {
+				template: 'a',
+				useNative: false,
+				timezone: 'local'
+			};
+
+			if (this.locale) fmtParams.locale = this.locale;
+
+			var merFormatter = new ilib.DateFmt(fmtParams);	
+			this.meridiems = merFormatter.getMeridiemsRange(fmtParams);
+		},
+
+		/**
+		* @private
+		*/
+		create: function() {
+			this.initILib();
```

Comment:
I think for the sake of consistency in 2.5 (we've made `enyo-ilib` a dependency of Moonstone in 2.6, so this is not applicable for 2.6), we may want to guard this based on the existence of `ilib`, as we are doing here: https://github.com/enyojs/moonstone/blob/2.5-upkeep/source/DateTimePickerBase.js#L186. What do you think?